### PR TITLE
Lcra/update acl organization workspace scenario

### PR DESCRIPTION
**Pull request by Leopold-Cramer**

### CODE REVIEW

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 1]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
+                solution = solutionApiService.createSolution(organization.id!!, mockSolution())
+
+                workspace =
+                    workspaceApiService.createWorkspace(
+                        organization.id!!, mockWorkspace(roleName = FAKE_MAIL, role = role))
+
+                if (shouldThrow) {
+                  assertThrows<CsmAccessForbiddenException> {
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:786]: message validation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 2]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC setWorkspaceDefaultSecurity : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
+                solution = solutionApiService.createSolution(organization.id!!, mockSolution())
+
+                every { getCurrentAuthenticatedRoles(any()) } returns listOf(role)
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:814]: remove

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 3]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                every { getCurrentAccountIdentifier(any()) } returns FAKE_MAIL
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:815]: move under workspace creation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 4]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_VALIDATOR to true,
+                ROLE_USER to true,
+                ROLE_NONE to true,
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC addWorkspaceAccessControl : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:849]: role

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 5]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_VALIDATOR to true,
+                ROLE_USER to true,
+                ROLE_NONE to true,
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC setWorkspaceDefaultSecurity : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:811]: role

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 6]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC addWorkspaceAccessControl : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
+                solution = solutionApiService.createSolution(organization.id!!, mockSolution())
+
+                every { getCurrentAuthenticatedRoles(any()) } returns listOf(role)
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:852]: remove

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 7]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                every { getCurrentAccountIdentifier(any()) } returns FAKE_MAIL
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:853]: move under workspace creation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 8]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+
+                workspace =
+                    workspaceApiService.createWorkspace(
+                        organization.id!!, mockWorkspace(roleName = FAKE_MAIL, role = role))
+
+                if (shouldThrow) {
+                  assertThrows<CsmAccessForbiddenException> {
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:860]: message validation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 9]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_VALIDATOR to true,
+                ROLE_USER to false,
+                ROLE_NONE to true,
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC getWorkspaceAccessControl : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:887]: role

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 10]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                solution = solutionApiService.createSolution(organization.id!!, mockSolution())
+
+                every { getCurrentAuthenticatedRoles(any()) } returns listOf(role)
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:890]: remove

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 11]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                every { getCurrentAccountIdentifier(any()) } returns FAKE_MAIL
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:891]: move under workspace creation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 12]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+
+                workspace =
+                    workspaceApiService.createWorkspace(
+                        organization.id!!, mockWorkspace(roleName = FAKE_MAIL, role = role))
+
+                if (shouldThrow) {
+                  assertThrows<CsmAccessForbiddenException> {
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:898]: message validation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 13]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC removeWorkspaceAccessControl : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
+                solution = solutionApiService.createSolution(organization.id!!, mockSolution())
+
+                every { getCurrentAuthenticatedRoles(any()) } returns listOf(role)
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:928]: remove

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 14]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                every { getCurrentAccountIdentifier(any()) } returns FAKE_MAIL
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:929]: move under workspace creation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 15]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+
+                workspace =
+                    workspaceApiService.createWorkspace(
+                        organization.id!!, mockWorkspace(roleName = FAKE_MAIL, role = role))
+
+                if (shouldThrow) {
+                  assertThrows<CsmAccessForbiddenException> {
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:936]: message validation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 16]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_VALIDATOR to true,
+                ROLE_USER to true,
+                ROLE_NONE to true,
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC updateWorkspaceAccessControl : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:963]: role

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 17]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                solution = solutionApiService.createSolution(organization.id!!, mockSolution())
+
+                every { getCurrentAuthenticatedRoles(any()) } returns listOf(role)
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:966]: remove

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 18]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                every { getCurrentAccountIdentifier(any()) } returns FAKE_MAIL
+
+                workspace =
+                    workspaceApiService.createWorkspace(
+                        organization.id!!, mockWorkspace(roleName = FAKE_MAIL, role = role))
+
+                if (shouldThrow) {
+                  assertThrows<CsmAccessForbiddenException> {
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:974]: message validation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 19]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_VALIDATOR to true,
+                ROLE_USER to false,
+                ROLE_NONE to true,
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC getWorkspaceSecurityUsers : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1001]: role

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 20]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                solution = solutionApiService.createSolution(organization.id!!, mockSolution())
+
+                every { getCurrentAuthenticatedRoles(any()) } returns listOf(role)
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1004]: remove

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 21]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                every { getCurrentAccountIdentifier(any()) } returns FAKE_MAIL
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1005]: move under workspace creation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 22]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+
+                workspace =
+                    workspaceApiService.createWorkspace(
+                        organization.id!!, mockWorkspace(roleName = FAKE_MAIL, role = role))
+
+                if (shouldThrow) {
+                  assertThrows<CsmAccessForbiddenException> {
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1012]: message validation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 23]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_VALIDATOR to true,
+                ROLE_USER to false,
+                ROLE_NONE to true,
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC createSecret : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1037]: role

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 24]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                solution = solutionApiService.createSolution(organization.id!!, mockSolution())
+
+                every { getCurrentAuthenticatedRoles(any()) } returns listOf(role)
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1040]: remove

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 25]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                every { getCurrentAccountIdentifier(any()) } returns FAKE_MAIL
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1041]: move under workspace creation

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 26]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                ROLE_VALIDATOR to false,
+                ROLE_USER to false,
+                ROLE_NONE to false,
+                ROLE_ADMIN to false,
+            )
+            .map { (role, shouldThrow) ->
+              DynamicTest.dynamicTest("Test RBAC importWorkspace : $role") {
+                organization =
+                    organizationApiService.registerOrganization(
+                        mockOrganization(id = "id", roleName = FAKE_MAIL))
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1075]: role

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 27]
```diff
@@ -327,35 +319,769 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
...
+                solution = solutionApiService.createSolution(organization.id!!, mockSolution())
+
+                every { getCurrentAccountIdentifier(any()) } returns FAKE_MAIL
+
+                assertDoesNotThrow {
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1080]: add **assertThrows** and validation on exception message

File: workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt [hunk 28]
```diff
@@ -386,9 +1112,9 @@ class WorkspaceServiceIntegrationTest : CsmRedisTestBase() {
   }
 
   fun mockWorkspace(
-      organizationId: String,
-      solutionId: String,
-      name: String,
+      organizationId: String = organization.id!!,
+      solutionId: String = solution.id!!,
+      name: String = "name",
       roleName: String = CONNECTED_ADMIN_USER,
```
Comment (lalepee) [inline workspace/src/integrationTest/kotlin/com/cosmotech/worskpace/service/WorkspaceServiceIntegrationTest.kt:1118]: replace roleName with userName

**Review by lalepee — CHANGES_REQUESTED** [review]

File: dataset/src/integrationTest/kotlin/com/cosmotech/dataset/service/DatasetServiceRBACTest.kt [hunk 29]
```diff
@@ -0,0 +1,3 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.dataset.service
```
Comment (lalepee) [inline dataset/src/integrationTest/kotlin/com/cosmotech/dataset/service/DatasetServiceRBACTest.kt:3]: what is this ?

File: gradle.properties [hunk 30]
```diff
@@ -12,5 +12,3 @@ org.gradle.parallel=true
 org.gradle.jvmargs=-Xmx2g
 
 detekt.ignoreFailures=false
-
-useLocalLicenseAllowedFile=false
```
Comment (lalepee) [inline gradle.properties:16]: are you sure about that ?

**Review by lalepee — APPROVED** [review]

File: gradle.properties (hunk 30 above)
Comment (Leopold-Cramer) [reply gradle.properties:16]: YES

File: dataset/src/integrationTest/kotlin/com/cosmotech/dataset/service/DatasetServiceRBACTest.kt (hunk 29 above)
Comment (Leopold-Cramer) [reply dataset/src/integrationTest/kotlin/com/cosmotech/dataset/service/DatasetServiceRBACTest.kt:3]: A mistake

**Pull request merged**